content/blog/how-to-create-a-ci-cd-pipeline-to-execute-automation-test-suite-with-hpe-greenlake-for-containers.md

```diff
@@ -0,0 +1,139 @@
+          name: "Execute cluster metrics gathering test suite"
+          command: xvfb-run katalonc -consoleLog -browserType="Chrome" -retry=0 -statusDelay=15 -testSuitePath="Test Suites/ClusterMetricsSuite" -executionProfile='default' -projectPath='/project/sample.prj' --config -webui.autoUpdateDrivers=true
+      - store_test_results:
+          path: ./tests/katalon/Reports
+      - store_artifacts:
+          path: /tmp/project/
+workflows:
+  cluster-observability:
+    triggers:
+      - schedule:
+          cron: "0 20 * * *"
+          filters:
+            branches:
+              only:
+                - master
+    jobs:
+      - checkout-workspace
+      - copyright-check
+      - gather-cluster-metrics
+          requires:
+            - checkout-workspace
+      - cluster-scale-operations
+          requires:
+            - checkout-workspace          
+      - update-results
+          requires:
+            - checkout-workspace 
+```
+
+**Sample CircleCI workflow can be demonstrated as below:** 
+
+![Sample CircleCI Automation pipeline](/img/circleci-workflow.jpg "Sample CircleCI Automation pipeline")
+
+**A csvDataSource.csv artifact file may look like what's shown below:**
+
+```sql
+IDCluster,DateTime,BlueprintType,CPUUsage,MemoryAllocation,StorageCapacity,WorkerNodeCount,ScalingRequired,ClusterScaleUpDuration,ClusterScaleDownDuration,ClusterCreationDuration,ClusterDeletionDuration
+2a0810e6-0c32-451b-933b-74fbdf86358a,2022-06-15 20:00:00,standard,50%,70%,5GB,3,N,,,,
+ce18d4e0-9af3-40da-8d43-266fe05d17ba,2022-06-16 20:00:00,standard,80%,70%,5GB,3,Y,01,,,
+767801bd-7f1e-4a9a-804e-b560f168d968,2022-06-17 20:00:00,standard,10%,10%,5GB,4,Y,,01,,
+r3b185d5-c96a-49a5-b6de-13ae93c93fd4,2022-06-18 20:00:00,standard,07%,10%,5GB,2,N,,,,
+```
+
+As illustrated in the above data file, the UI automation run gathers data for the cluster where the application is deployed. A threshold value can be predefined for scaling requirements in the Katalon script. For example, 80% above usage should trigger scaling up, while 10% usage should trigger scaling down. As per the first record after app deployment, CPU and Memory metrics are 50% and 70%, respectively, for which scaling is not required which is marked as 'N' in the data file. On the second day, the CPU reached 80% usage, and the required time for a scale operation to complete is 1 minute, which is also captured. 
+
+## How to Download Artifacts from CircleCI Workflow Dynamically?
+
+CircleCI provides [API Token](https://circleci.com/docs/2.0/managing-api-tokens/#creating-a-personal-api-token) to view pipelines via an API interface. Based upon requirements, various [APIs ](https://circleci.com/docs/api/v2/)can be selected. In the [postman ](https://www.postman.com/downloads/)tool, a user can try a combination of APIs for passing the output of one API result to the input of another API. A curl command for downloading artifacts from CircleCI API may look like the one mentioned below:
```

Comment:
A suggested rewrite of the beginning of the first sentence here:
"CircleCI **offers** [API Token]..."

(when one says CircleCI provides API Token, it sounds as if one should say "an API Token", but I don't think that's what you really mean. "Offers" gets us around that issue...)

I believe "Postman" should have an initial capital "P"